### PR TITLE
[#2136] feat:(Server/Coordinator) Support a way to updateConf from rest api

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes were proposed in this pull request?

Support a way to updateConf from rest api

### Why are the changes needed?

To support dynamic update conf.

### Does this PR introduce _any_ user-facing change?

Yes, supply a rest api.

### How was this patch tested?

Tested Locally.

```Console
curl -X POST http://localhost:19978/api/shuffleServer/confOps/update -d "key=rss.server.log.app.detail.while.remove.enabled" -d "value=true"
Set key: rss.server.log.app.detail.while.remove.enabled from 'null' to 'true'%
```
```
[2024-09-23 23:00:13.715] [Jetty-5] [INFO] ConfOpsResource - Dynamic updating rss.server.log.app.detail.while.remove.enabled to true, delete=false
```

### CODE REVIEW

**Comment by jerqi** [comment]
Why do u use REST API instead of rpc?

**Comment by maobaolong** [comment]
@jerqi Since we have no rpc based client cli, AFAIK, the uniffle cli is based rest api for now?

**Comment by zuston** [comment]
I fully understand the motivation of this feature, but this conf update only occurs in the current lifecycle of shuffle-server. If it restart, the updated conf will lost.

For a large uniffle cluster, this may increases the probability of erroneous operation, although this is not the fault of this feature

**Comment by maobaolong** [comment]
@zuston yep, this update is not persistent behavior, but a way to update conf without need to restart or conf file modification.

If user use this rest api, it make him convenient but user should know the config will lost when restart.

So, I can add a warn content to user when user invoke this rest api.

File: common/src/main/java/org/apache/uniffle/common/web/resource/ConfOpsResource.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.uniffle.common.config.RssConf;
+
+@Path("/confOps")
+public class ConfOpsResource {
+  private static final Logger LOG = LoggerFactory.getLogger(ConfOpsResource.class);
+  public static final String SERVLET_CONTEXT_ATTR_CONF = "_servlet_context_attr_conf_";
+  private static final String WARNING_MSG = "WARN: The change will lost after restart.\n";
```
[the remark this answers is not in the document]
Comment (maobaolong) [reply common/src/main/java/org/apache/uniffle/common/web/resource/ConfOpsResource.java:37]: done

File: common/src/main/java/org/apache/uniffle/common/web/resource/ConfOpsResource.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          conf.remove(key);
+          return WARNING_MSG + "Removed key: " + key;
+        } else {
+          String oldValue = conf.getString(key, null);
+          conf.setString(key, value);
+          return WARNING_MSG + "Set key: " + key + " from '" + oldValue + "' to '" + value + "'";
+        }
+      }
+    }
+    return response;
```
[the remark this answers is not in the document]
Comment (maobaolong) [reply common/src/main/java/org/apache/uniffle/common/web/resource/ConfOpsResource.java:63]: Done

File: common/src/main/java/org/apache/uniffle/common/web/resource/ConfOpsResource.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  @Authorization
+  @POST
+  @Path("/update")
+  public String update(
+      @FormParam("key") String key,
+      @FormParam("value") String value,
+      @FormParam("delete") @DefaultValue("false") boolean delete) {
+    LOG.info("Dynamic updating {} to {}, delete={}", key, value, delete);
+    String response = "do nothing";
```
[the remark this answers is not in the document]
Comment (maobaolong) [reply common/src/main/java/org/apache/uniffle/common/web/resource/ConfOpsResource.java:49]: Done

File: common/src/main/java/org/apache/uniffle/common/web/resource/ConfOpsResource.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    if (Strings.isNotEmpty(key)) {
+      RssConf conf = (RssConf) servletContext.getAttribute(SERVLET_CONTEXT_ATTR_CONF);
+      if (conf != null) {
+        if (delete) {
+          conf.remove(key);
+          return WARNING_MSG + "Removed key: " + key;
+        } else {
+          String oldValue = conf.getString(key, null);
+          conf.setString(key, value);
+          return WARNING_MSG + "Set key: " + key + " from '" + oldValue + "' to '" + value + "'";
```
[the remark this answers is not in the document]
Comment (maobaolong) [reply common/src/main/java/org/apache/uniffle/common/web/resource/ConfOpsResource.java:59]: done

[the remark this answers is not in the document]
Comment (maobaolong) [reply common/src/main/java/org/apache/uniffle/common/web/resource/ConfOpsResource.java:59]: done

File: common/src/main/java/org/apache/uniffle/common/web/resource/ConfOpsResource.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      @FormParam("value") String value,
+      @FormParam("delete") @DefaultValue("false") boolean delete) {
+    LOG.info("Dynamic updating {} to {}, delete={}", key, value, delete);
+    String response = "do nothing";
+    if (Strings.isNotEmpty(key)) {
+      RssConf conf = (RssConf) servletContext.getAttribute(SERVLET_CONTEXT_ATTR_CONF);
+      if (conf != null) {
+        if (delete) {
+          conf.remove(key);
+          return WARNING_MSG + "Removed key: " + key;
```
[the remark this answers is not in the document]
Comment (maobaolong) [reply common/src/main/java/org/apache/uniffle/common/web/resource/ConfOpsResource.java:55]: done

File: dashboard/src/main/java/org/apache/uniffle/dashboard/web/resource/DashboardResource.java [hunk 6]
```diff
@@ -72,6 +73,11 @@ public Response<Map<String, Object>> getDashboardInfo() {
         });
   }
 
+  @Path("/confOps")
+  public Class<ConfOpsResource> getConfOps() {
```
[the remark this answers is not in the document]
Comment (maobaolong) [reply dashboard/src/main/java/org/apache/uniffle/dashboard/web/resource/DashboardResource.java:77]: Thanks for your remind. Revert the change related to Dashboard and updated the title of this PR.

**Comment by maobaolong** [comment]
@advancedxy Thanks for your review and suggested change, I've addressed it all, PTAL.

**Comment by maobaolong** [comment]
@advancedxy Yes, so many PR troubled by that test, I submitted a PR to disable that UT by https://github.com/apache/incubator-uniffle/pull/2151

**Review by zuston — APPROVED** [review]
+1.

**Pull request merged**